cmd/internal/kube/inject.go

```diff
@@ -28,13 +32,41 @@ var (
 	// When set to "true", injectCmd will prepend a secret to each injectable namespace found in the file to inject (injectFileNameFlag)
 	// Otherwise, injectCmd will treat secretInjectFlag as the file path all secrets should be generated to
 	secretInjectFlag string
+
+	// Postman related flags
+	postmanCollectionID string
+	postmanEnvironment  string
 )
 
 var injectCmd = &cobra.Command{
 	Use:   "inject",
 	Short: "Inject Akita into a Kubernetes deployment",
 	Long:  "Inject Akita into a Kubernetes deployment or set of deployments, and output the result to stdout or a file",
 	RunE: func(_ *cobra.Command, args []string) error {
+		if projectNameFlag == "" && postmanCollectionID == "" {
+			return cmderr.AkitaErr{
+				Err: errors.New("exactly one of --project or --collection must be specified"),
+			}
+		}
+
+		// If --collection was given, set rest.domain and environment config (if given)
+		if postmanCollectionID != "" {
+			if postmanEnvironment != "" {
+				env := strings.ToUpper(postmanEnvironment)
+
+				cfg.WritePostmanEnvironment("default", env)
+			}
+
+			rest.Domain = "staging.akita.software"
+		}
+
+		// Lookup service *first* (if we are remote) ensuring that collectionId
+		// or projectName is correct and exists.
+		err := lookupService(postmanCollectionID, projectNameFlag)
+		if err != nil {
+			return err
+		}
```

Comment:
I don't recall what was behind @versilis 's decision _not_ to make this check.  Perhaps we wanted `kube inject` to work even in environments where the API endpoint was unreachable.  

@versilis do you recall if there was a requirement or use case here?

Reply:
As discussed will be keeping this check now